### PR TITLE
feat: add GCS as a FileStore

### PR DESCRIPTION
**Pull request by mattkrick**
# Description

Fixesx #8251 

Allows you to use GCS as a file store provider 🎉 

## Demo

Screenshot 2023-07-07 at 2 41 22 PM

## Testing scenarios

- [ ] in your .env set `GOOGLE_GCS_BUCKET='parabol-1'` and `FILE_STORE_PROVIDER='gcs'` and `CDN_BASE_URL=// storage.googleapis.com/parabol-1/development`
- [ ] change your profile pic & see it upload 🚀 
- [ ] run `node dist/pushToCDN` and it will attempt to upload any new templates
- [ ] you can also test this with s3 (no DNS changes necessary) by setting `CDN_BASE_URL=//s3.amazonaws.com/action-files.parabol.co/development`
- [ ] if you change the DNS record for action-files.parabol.co, you can set `CDN_BASE_URL=//action-files.parabol.co`, but that's optional

### CODE REVIEW

**Comment by rafaelromcar-parabol** [comment]
Sorry for the time it took to test it.

I'm testing it in an on-demand environment (k8s based) from the image generated from the last commit I pushed (adding a dummy text file) and I'm getting the following:

```
[node@parabol-predeploy-9n4tj parabol]$ cat .env
#
# Required arguments
#
FILE_STORE_PROVIDER='gcs'
CDN_BASE_URL='//assets.parabol-saas-development.parabol.fun/'
GOOGLE_GCS_BUCKET='parabol-saas-development-assets'
PORT=3000
PROTO='https'
SOCKET_PORT='3001'
MAIL_PROVIDER='mailgun'
MAIL_FROM='parabol@mail.parabol.co'
#CHRONOS_PULSE_EMAIL=''
#CHRONOS_PULSE_CHANNEL=''
#CHRONOS_PULSE_DAILY='0 0 4 * * *'
#CHRONOS_PULSE_WEEKLY='0 0 4 * * 1'
#CHRONOS_BATCH_EMAILS='0 0 10 * * *'
#CHRONOS_SCHEDULE_JOBS='0 */10 * * * *'
#CHRONOS_UPDATE_TOKENS='0 0 0 1,15 * *'
#CHRONOS_PROCESS_RECURRENCE='0 */5 * * * *'
#CHRONOS_AUTOPAUSE='0 0 5 * * *'
DD_SERVICE='parabol-local-dev'
DD_TRACE_ENABLED='false'
DD_APPSEC_ENABLED='false'
DD_TRACE_AGENT_PORT=8126
DD_APM_ENABLED='false'
DD_LOGS_INJECTION='true'
DD_RUNTIME_METRICS_ENABLED='true'
DD_TRACE_ANALYTICS_ENABLED='true'
DD_ENV='local'

[node@parabol-predeploy-9n4tj parabol]$ export SERVER_ID=202 && node dist/preDeploy.js
👴 RethinkDB Migraiton Started
[migrate-rethinkdb] No new migrations
👴 RethinkDB Migraiton Complete
🐘 Postgres Migraiton Started
Post deploy error Error: Mssing Env: FILE_STORE_PROVIDER
    at _ (/home/node/parabol/dist/preDeploy.js:1:889639)
    at /home/node/parabol/dist/preDeploy.js:1:890056
    at Array.map (<anonymous>)
    at 86785 (/home/node/parabol/dist/preDeploy.js:1:889851)
    at n (/home/node/parabol/dist/preDeploy.js:1:6325855)
    at i (/home/node/parabol/dist/preDeploy.js:1:5839536)
    at /home/node/parabol/dist/preDeploy.js:1:913359
    at Array.forEach (<anonymous>)
    at /home/node/parabol/dist/preDeploy.js:1:913258
    at u (/home/node/parabol/dist/preDeploy.js:1:913611)
```

As you can see, there is .env file with the variable `FILE_STORE_PROVIDER` present. Even if I force and add it exporting it right before I get the following:

```
[node@parabol-predeploy-9n4tj parabol]$ export FILE_STORE_PROVIDER='gcs' && export SERVER_ID=202 && node dist/preDeploy.js
👴 RethinkDB Migraiton Started
[migrate-rethinkdb] No new migrations
👴 RethinkDB Migraiton Complete
🐘 Postgres Migraiton Started
Post deploy error Error: Mssing Env: FILE_STORE_PROVIDER
    at _ (/home/node/parabol/dist/preDeploy.js:1:889639)
    at /home/node/parabol/dist/preDeploy.js:1:890056
    at Array.map (<anonymous>)
    at 86785 (/home/node/parabol/dist/preDeploy.js:1:889851)
    at n (/home/node/parabol/dist/preDeploy.js:1:6325855)
    at i (/home/node/parabol/dist/preDeploy.js:1:5839536)
    at /home/node/parabol/dist/preDeploy.js:1:913359
    at Array.forEach (<anonymous>)
    at /home/node/parabol/dist/preDeploy.js:1:913258
    at u (/home/node/parabol/dist/preDeploy.js:1:913611)
```

It doesn't make much sense 🤔 WDYT?

Also, **could you rebase the branch on master and generate a new docker image after that please?** That would already help making the predeploy exit with code non-zero when it is failing 😸 (I know it was my fault too). I'll get back to this on Wednesday morning. If I'm not done when you are online on Wednesday, we will do a sync and end this on that day.

File: packages/server/postgres/migrations/1685721573097_addPrePostMortemTemplates.ts [hunk 1]
```diff
@@ -982,13 +982,12 @@ const getTemplateIllustrationUrl = (filename: string) => {
   const partialPath = `Organization/aGhostOrg/${filename}`
   if (cdnType === 'local') {
     return `/self-hosted/${partialPath}`
-  } else if (cdnType === 's3') {
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1685721573097_addPrePostMortemTemplates.ts:985]: i don't like mutating past migrations, but this logic doesn't change anything, it just adds support for gcs & any future providers

**Comment by mattkrick** [comment]
@rafaelromcar-parabol ready for review!

**Comment by mattkrick** [comment]
@igorlesnenko since Rafa is all tied up with moving to K8S would you mind reviewing this? Ideally we'll be moving from S3 to GCS in the next couple weeks

**Review by igorlesnenko — APPROVED** [review]
Works! Everything looks good!

Just need to run `yarn` again

**Pull request merged**